design/one-pager-stack-status-spec.md

```diff
@@ -0,0 +1,114 @@
+# Stack UI Schema Status Spec
+
+- Owner: Steven Rathbauer (@rathpc)
+- Reviewers: Crossplane Maintainers
+- Status: Draft
+
+## Proposal
+
+[crossplaneio/crossplane#929](https://github.com/crossplaneio/crossplane/issues/929)
+
+## Problem
+
+For a given CRD I want to be able to expose various statuses to a frontend GUI. Initially it makes
+sense to at least have the Ready state available. In addition to the value of the Ready state it
+would also be helpful to have some additional details about that state if applicable.
+
+This also assumes that statuses are following the podCondition object convention as described here:
```

Comment:
I think something I'm struggling with here is that not everything is a pod nor will they have the same semantics that pod conditions have.  In the core k8s types, there are different condition types for different API types.  Here are a few examples:

* `PersistentVolumeClaimCondition`: https://github.com/kubernetes/api/blob/master/core/v1/types.go#L487
* `NodeCondition`: https://github.com/kubernetes/api/blob/master/core/v1/types.go#L4511
* `ComponentCondition`: https://github.com/kubernetes/api/blob/master/core/v1/types.go#L5570

And each one of them has their own set of available condition types, e.g. [`ComponentConditionType`](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L5566) has a totally separate set of types than [`NodeConditionType`](https://github.com/kubernetes/api/blob/master/core/v1/types.go#L4497-L4508)

I think it's reasonable that for surfacing status we can standardize on conditions, and that those conditions can have `type`, `status`, `reason`, and `message` fields.  But I'm not sure how we can best standardize on the **semantics** of those conditions, i.e. what each one actually **means**.

One CRD may say it's ready with a `Ready` condition, while another CRD may say the same thing with `Available`.

What do you think about how we can standardize on **semantics** or **meaning** of conditions in a world of arbitrary CRDs?

Reply:
Or, do we not even need to standardize on meaning of conditions? maybe that doesn't have to be a goal?  A GUI could blindly display the conditions that are surfaced without any understanding of its meaning, if we wanted to go that way.

Reply:
I will be updating this to reference the following of a convention rather than a specific type